From: Seth Heasley <seth.heasley@intel.com>
Subject: [PATCH] x86/PCI: irq and pci_ids patch for additional Intel Cougar Point DeviceIDs
Patch-mainline: 33852cb03ee4cdb05dc6e3a21ec19a4ee63511a4
References: bnc#708269

This patch adds additional LPC Controller DeviceIDs for the Intel Cougar
Point PCH.

The DeviceIDs are defined and referenced as a range of values, the same
way Ibex Peak was implemented.

Signed-off-by: Seth Heasley <seth.heasley@intel.com>
Signed-off-by: Jesse Barnes <jbarnes@virtuousgeek.org>
Acked-by: Jiri Kosina <jkosina@suse.cz>
Automatically created from "patches.drivers/intel-additional-Intel-Cougar-Point.patch" by xen-port-patches.py

Index: linux-2.6.32-SLE11-SP2/arch/x86/pci/irq-xen.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/arch/x86/pci/irq-xen.c
+++ linux-2.6.32-SLE11-SP2/arch/x86/pci/irq-xen.c
@@ -590,8 +590,6 @@ static __init int intel_router_probe(str
 	case PCI_DEVICE_ID_INTEL_ICH10_1:
 	case PCI_DEVICE_ID_INTEL_ICH10_2:
 	case PCI_DEVICE_ID_INTEL_ICH10_3:
-	case PCI_DEVICE_ID_INTEL_CPT_LPC1:
-	case PCI_DEVICE_ID_INTEL_CPT_LPC2:
 	case PCI_DEVICE_ID_INTEL_PATSBURG_LPC_1:
 	case PCI_DEVICE_ID_INTEL_PATSBURG_LPC_2:
 		r->name = "PIIX/ICH";
@@ -610,6 +608,13 @@ static __init int intel_router_probe(str
 		return 1;
 	}
 
+	if ((device >= PCI_DEVICE_ID_INTEL_CPT_LPC_MIN) && 
+		(device <= PCI_DEVICE_ID_INTEL_CPT_LPC_MAX)) {
+		r->name = "PIIX/ICH";
+		r->get = pirq_piix_get;
+		r->set = pirq_piix_set;
+		return 1;
+	}
 	return 0;
 }
 
